### ISSUE TITLE
CloudCompare v2.12.alpha - Executing command's via Phyton subprocess model result's in returned non-zero exit status 3221226505

### ISSUE THREAD

**Issue by DavixDevelop** [CloudCompare/CloudCompare#1465]
**Describe the bug**
When using the command line mode via cmd, it works as it should, but if you for example use a module like the subprocess module in Python to execute the command it returns an error:
` Command <command here> returned non-zero exit status 3221226505`

To be precise, this error was returned when using the `subprocess.check_output` function. 
I did beforehand search for solutions online and found out that this particular error also happened in the past, so I tried the various solutions to the problem, like changing the current working directory to where CloudComapre is installed via `os.chdir`, via the `Popen` constructor or by specifying the `cwd` parameter in `subprocess.check_output`. I also tried the various combinations with setting the `shell` parameter to `False` or `True`, but none of them worked. 

The error only happens when using the v2.12.alpha version, and not in prior ones like v2.11.3, where the command always executes as it should via Phyton.

**How to reproduce**
1. Write a Python script where you use the `subprocess` module to launch a command for the CloudCompare command line mode, ex.
```phyton
subprocess.check_output('\"{cloud}\" -SILENT -O -GLOBAL_SHIFT AUTO  \"{file}\" -RASTERIZE -GRID_STEP 1 -VERT_DIR 2 -PROJ MIN -SF_PROJ AVG -EMPTY_FILL INTERP -OUTPUT_RASTER_Z'.format(cloud=job_data.cloudCompare,file=file_data.filePath), cwd=job_data.srcDir, shell=True)
```
2. Put the line in a `try except` block and print out the caught error
3. Execute the script via the terminal or via a python console like in QGIS 
4. The script print's out `Command <command here> returned non-zero exit status 3221226505`

**Expected behavior**
Executing the script should launch the CloudCompare command line mode and execute the task as it did in previous versions of CloudCompare.

**Additional context**
I wrote a Python script that people use in QGIS to rasterize a folder of point cloud's for a project I'm part of, and a such when I noticed that v2.12.alpha is available I updated the guide on our wiki, without checking if the script still works, when people in the project notified me that it, in fact, executing the script via QGIS doesn't work anymore. Then I tried the various fixes, thinking something was wrong with my script, but when I downgraded to the previous version (v2.11.3) everything worked as before.

**Your environment**
 - CC Version (or git hash/tag): CloudCompare v2.12.alpha
 - OS & Version: Windows 10 Pro - 20H2
 - Graphics card: NVIDIA GeForce GTX 1050 Ti with Max-Q Design
 - QGIS Version: QGIS 3.18
 - Python Version: Python 3.7.0

**Comment by dgirardeau** [comment]
Is it a DLL issue maybe? A dependency to something that the way you (or Python) launch CC prevents it from finding the right DLL?

Have you installed the FARO plugin maybe? (this one is a strange beast for instance)

**Comment by DavixDevelop** [comment]
No, I haven't installed the FARO plugin. I and the user who reported the problem just uninstalled v2.11.3 and installed v2.12.alpha and the error showed up. I can launch any other executable with no problem via `subprocess.check_output`, so the script definitely isn't a problem, as it works with the v2.11.3 version. Also, I think if it couldn't find the right DLL, setting the current working directory to where CloudCompare is installed, should have installed resolved the problem, but it didn't, so I really don't know what could be the problem.

**Comment by tmontaigu** [comment]
I'm able to reproduce when following tutorial given and running export_dem.py with the QGIS's Python.

However I extracted the cloudcompare calls to another script, to run it independently from QGIS
```Python
import subprocess
from pathlib import Path

cloud_compare_path = Path(r"~\Desktop\CloudCompare_v2.12.alpha_bin_x64\CloudCompare.exe").expanduser()
file_path = Path(r"~\Desktop\003_BC_04_0+000_0+900_8 - Cloud.bin").expanduser()

extract_ground = False  # Set it to True if your LiDAR data contains both ground and off-ground points (Like trees, buildings...)
recursive = False  # Set to True if you want the script to also scan for files in sub-folders
thread_count = None  # Set to the number of threads you want to use. Preferably don't use all threads at once. Leave at at None to use all threads
default_nodata = -32768  # Don't change the value of nodata, if you don't know it

subprocess.check_output(
    '\"{cloud}\" -SILENT -O -GLOBAL_SHIFT AUTO  \"{file}\" -RASTERIZE -GRID_STEP 0.25 -VERT_DIR 2 -PROJ MIN -SF_PROJ AVG -EMPTY_FILL INTERP -OUTPUT_RASTER_Z'.format(
        cloud=cloud_compare_path,
        file=file_path),
    shell=True)

subprocess.check_output(
    '\"{cloud}\" -SILENT -O -GLOBAL_SHIFT AUTO  \"{file}\" -CSF -SCENES SLOPE -CLOTH_RESOLUTION 0.5 -CLASS_THRESHOLD 0.5 -RASTERIZE -GRID_STEP 0.25 -VERT_DIR 2 -PROJ MIN -SF_PROJ AVG -EMPTY_FILL INTERP -OUTPUT_RASTER_Z'.format(
        cloud=cloud_compare_path,
        stderr=subprocess.STDOUT,
        file=file_path),
    shell=True)
```

And CloudCompare runs fine.

So the problems comes from running CloudCompare and QGIS, my guess would be that the cloudcompare process started inherits some of the properties of the QGIS process and that creates the problem.
 
Maybe it inherits some DLLs that QGIS already loaded and CloudCompare needs, for example Qt, QGis already loaded Qt5*.dll,
CloudCompare also wants to load a Qt5*.dll, but as its already loaded it use the one loaded by QGIS, not the ones it bundles.

QGIS 3.10 uses Qt5.12, CC v2.12alpha is compiled against Qt5.15,  while CC 2.11 uses Qt 5.9.9,  so this might create an incompatibility at runtime, at least that's what I'm speculating.

**Comment by sladg** [comment]
I'm experiencing same behaviour when using Windows and NodeJS.
Running `CloudCompare.exe -SILENT -LOG_FILE ./cc.log -DEBUG` works in cmd itself, however, running `child_process.exec` in Node does not. It exists with Code -1 without obvious cause.

(works fine on Linux)
